src/dmd/compiler.d

```diff
@@ -12,7 +12,12 @@
 
 module dmd.compiler;
 
+import dmd.globals;
+
 struct Compiler
 {
-    const(char)* vendor; // Compiler backend name
+    extern(C++) static void _init()
+    {
+        global.vendor = "Digital Mars D";
+    }
 }
```

Comment:
As discussed on IRC: how about making `Compiler` an interface instead?

Reply:
I've given it some thought and I'd rather avoid this if possible.  Having to maintain interfaces seems like a last resort solution.   For sure though `Global._init` is a _big_ problem to tackle, as a big chunk of it should go into `Target`, but the globals module cannot import target!

I'll start off with adding `initGlobals` and see where we go from there.